bee-block/src/output/basic.rs

```diff
@@ -131,36 +130,37 @@ impl BasicOutputBuilder {
     }
 
     ///
-    pub fn finish(self) -> Result<BasicOutput, Error> {
+    pub fn finish(self, protocol_parameters: &ProtocolParameters) -> Result<BasicOutput, Error> {
```

Comment:
Should we really accept the whole protocol_parameters, even when we just need to have the max supply and rent structure? Looks like it could be annoying to always fill all other fields in the ProtocolParameters, when they aren't even needed

Reply:
This is something I asked myself. From the bee point of view I don't think it's an issue since it's really low level. For the client, we can get the whole struct from a getInfo call so it shouldn't be too annoying?

Reply:
Yes when we have the nodeinfo already available it shouldn't be annoying, but I think there could be some cases where this basically adds the need to get the nodeinfo now, where it wasn't required before
For example https://github.com/iotaledger/iota.rs/blob/develop/examples/outputs/basic.rs would require an additional nodeinfo call before we build the outputs

Reply:
And how do you get the supply for the specific network you are interested in ?

Reply:
Could be hardcoded if you know that you're only working with a specific network